### ISSUE TITLE
Mark SUComponentDefinitionGetInsertPoint	and SUComponentDefinitionSetInsertPoint as deprecated (noop.)

### ISSUE THREAD

**Issue by DanRathbun** [SketchUp/api-issue-tracker#451]
#### SketchUp C API Issue

[`SUComponentDefinitionGetInsertPoint`](https://extensions.sketchup.com/developers/sketchup_c_api/sketchup/component__definition_8h.html#abc02a3d9fcf123906c78446a9ebb8f52) and [`SUComponentDefinitionSetInsertPoint`](https://extensions.sketchup.com/developers/sketchup_c_api/sketchup/component__definition_8h.html#aed56436892063e5a7deb5d5568ee3a77) should be marked as deprecated (or noop) in 2020 C API documentation.

[`SUComponentDefinitionGetInsertPoint`](https://extensions.sketchup.com/developers/sketchup_c_api/sketchup/component__definition_8h.html#abc02a3d9fcf123906c78446a9ebb8f52) should just return the definitions origin (from an identity transform.)

Perhaps [`SUComponentDefinitionSetInsertPoint`](https://extensions.sketchup.com/developers/sketchup_c_api/sketchup/component__definition_8h.html#aed56436892063e5a7deb5d5568ee3a77) does nothing, and should it return an `SU_ERROR_UNSUPPORTED` error code for API ver **8** (SketchUp 2020) and later ?

**Comment by thomthom** [comment]
`SUComponentDefinitionSetInsertPoint` is already marked as deprecated.

`SUComponentDefinitionGetInsertPoint` appear to have slipped away from this tag. It does however just return the default origin.

**Comment by DanRathbun** [comment]
> `SUComponentDefinitionGetInsertPoint` appear to have slipped away from this tag. It does however just return the default origin.

Okay, but I cannot see what the doc says for this function as the doc outputs for v8 do not properly list the document for the functions. Some only the first function. Most classes no function docs at all. This calls only lists `SUComponentDefinitionCreateInstance` as the only function.

**Please check the source for malformed doxygen comments.**

> `SUComponentDefinitionSetInsertPoint` is already marked as deprecated.

Probably but the online docs have not yet been updated for the C API (for ver 8 / SU2020.)
I hadn't yet noticed when I opened this issue, I think.

What do you think about it's returning an error code?

**Comment by thomthom** [comment]
Yea, the C API docs has been a pain, as currently they only build on the build servers - not on local machine. This week I was looking into that and figured out how to do so - which finally makes it easier to pick up malformed syntax. (We got some cleanup work to do.)
I also found a Doxygen theme replacement which makes the docs a lot easier to navigate and (* gasp * functioning and useful search!)

**Comment by DanRathbun** [comment]
Cool. I hate how the functions are all mixed up alphabetically.

**Comment by thomthom** [comment]
> Cool. I hate how the functions are all mixed up alphabetically.

How do you men? Force grouping of `Get` and `Set` functions?

**Comment by DanRathbun** [comment]
**Not exactly**. Neither the short list nor the detail listings are in alpha order. It makes it tedious to find the function I'm looking for.

I know, _I know_ the global function list **is** in alpha order ...
https://extensions.sketchup.com/developers/sketchup_c_api/sketchup/globals_func.html
... but I always first just go to the class page we are discussing and go looking for the function.

The class pages seem to first list the export functions, then list the support functions in what looks like the definition order of the header file(s). (Actually they **are** in **exact** header file order.)

**Comment by thomthom** [comment]
Never noticed that.

You mean the file view? Like this?
https://extensions.sketchup.com/developers/sketchup_c_api/sketchup/face_8h.html

I think there are settings in Doxygen for this.

**Comment by DanRathbun** [comment]
> You mean the file view? Like this?

Yes. They are just in the header file order. (Probably the doxygen default.)

**Comment by thomthom** [comment]
Doxygen has a config option to sort by names or by declaration order.

Not sure of declaration order has been intended in the past. It does ensure that GetSet functions are listed close to each other. 

Will have to look closer as we revise the C API docs.

**Comment by DanRathbun** [comment]
> `SUComponentDefinitionSetInsertPoint` is already marked as deprecated.
> 
> `SUComponentDefinitionGetInsertPoint` appear to have slipped away from this tag. It does however just return the default origin.

Since this issue was opened, the C API docs for v8 (2020) were published and appear correctly.

----

With regard to `SUComponentDefinitionGetInsertPoint`, a note might be added to say ...

> _As of SketchUp 2020.0, this function will return the default origin `{0,0,0}`, as the insertion point can no longer be changed._

**But, in addition, what should be said about old components added in that has a non-origin insertion point ?**
